pkg/interceptors/cel/cel_test.go

```diff
@@ -0,0 +1,247 @@
+package cel
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	"github.com/tektoncd/pipeline/pkg/logging"
+	triggersv1 "github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	fakekubeclient "knative.dev/pkg/client/injection/kube/client/fake"
+	rtesting "knative.dev/pkg/reconciler/testing"
+)
+
+// Allow configuration via a config map
+
+func TestInterceptor_ExecuteTrigger(t *testing.T) {
+	type args struct {
+		payload []byte
+	}
+	tests := []struct {
+		name    string
+		CEL     *triggersv1.CELInterceptor
+		args    args
+		want    []byte
+		wantErr bool
+	}{
+		{
+			name: "simple body check with matching body",
+			CEL: &triggersv1.CELInterceptor{
+				Expression: "body.value == 'testing'",
+			},
+			args: args{
+				payload: []byte(`{"value":"testing"}`),
+			},
+			want:    []byte(`{"value":"testing"}`),
+			wantErr: false,
+		},
+		{
+			name: "simple body check with non-matching body",
+			CEL: &triggersv1.CELInterceptor{
+				Expression: "body.value == 'test'",
+			},
+			args: args{
+				payload: []byte(`{"value":"testing"}`),
+			},
+			wantErr: true,
+		},
+		{
+			name: "simple header check with matching header",
+			CEL: &triggersv1.CELInterceptor{
+				Expression: "headers['X-Test'][0] == 'test-value'",
+			},
+			args: args{
+				payload: []byte(`{}`),
+			},
+			want:    []byte(`{}`),
+			wantErr: false,
+		},
+		{
+			name: "simple header check with non matching header",
+			CEL: &triggersv1.CELInterceptor{
+				Expression: "headers['X-Test'][0] == 'unknown'",
```

Comment:
The `headers` part here is slightly different from `headers` in `TriggerBindings`. 
In Bindings, the keyword is `header` not `headers` -- though `headers` does sound like the more correct term.
Also, you do not have to use array accessors e.g.  `$(headers.foo)` not `$(headers.foo[0])`
Its probably worth being consistent here.

Reply:
Just to clarify, you would prefer it to emulate JSON path expressions in CEL?

Reply:
Yes!

Reply:
I'm wondering if CEL is the right expression language to use, maybe we could augment the JSON path expressions?

Reply:
To be clear, the syntax `header['X-Test'][0]` is one possible option.

The header value is a https://golang.org/pkg/net/http/#Header value, but, it's translated in CEL into a map, so, there's a custom match overload, which uses `.Get()` on the value, this is doing more than a simple key-value lookup.

I'm not sure how easy it is to emulate JSON path expressions in CEL without a significant amount of work.

Reply:
By "emulating", I was just referring to making it easy to access single valued headers i.e header['X-Test'][0] vs header['X-Test']. From reading @wlynch 's comment it seems like that might already be the case...
We definitely should not try to do anything more complicated than that 😅